feat(server/web): Initial support for RAF and SRW RAW formats

samip5 opened the pull request:
Fixes #1367
Fixes #1368

samip5 committed:
feat(server/web): Initial support for RAF and SRW RAW formats.

samip5 committed:
It should return the promise.

samip5 committed:
Better comment

samip5 committed:
feat(server/web): file-uploader needed changes.

samip5 committed:
feat(server/web): file-uploader needed changes.

alextran1502 commented on server/apps/microservices/src/processors/thumbnail.processor.ts:
```diff
@@ -49,11 +50,19 @@ export class ThumbnailGeneratorProcessor {
     if (asset.type == AssetType.IMAGE) {
       try {
-        await sharp(asset.originalPath, { failOnError: false })
+        await sharp(asset.originalPath, { failOnError: true })
```
If you change this to true, some of the conversions will flat out fail instead of converting to its best ability

samip5 replied:
But if it doesn't have that set, it flat out doesn't error and we cannot know it failed and cannot fallback to exiftool.

I'm open to ideas on how to handle that as sharp will only throw a generic error but not a custom one.

alextran1502 replied:
I don't know. The information that I have to work with is that if failOnError is set to false it will convert to its best ability, otherwise, it would throw an error.

So maybe you can keep this behavior as originally implemented and fall back in the catch block

samip5 replied:
> I don't know. The information that I have to work with is that if failOnError is set to false it will convert to its best ability, otherwise, it would throw an error.
> 
> So maybe you can keep this behavior as originally implemented and fall back in the catch block

It needs to throw an error as the .catch block will then let it the error be handled via exiftool. If exiftool errors for some reason, and the code is in the main catch block (and not in an `.catch`), I don't think anything would catch it?

samip5 replied:
It seems I was wrong about the `failOnError` and set it to false again. the `.catch` does get used with .srw and .raf pictures.

alextran1502 commented on server/apps/microservices/src/processors/thumbnail.processor.ts:
```diff
@@ -49,11 +50,19 @@ export class ThumbnailGeneratorProcessor {
+                asset.id +
+                ' using sharp, failing over to exiftool-vendored',
+            );
+            return exiftool.extractThumbnail(asset.originalPath, jpegThumbnailPath);
```
We also need to catch error in here I believe

jrasm91 replied:
This is awaited, so it'll be caught in the surrounding try/catch.

samip5 committed:
Merge branch 'immich-app:main' into feat/samsung-raw-and-fujifilm-raf

samip5 committed:
Merge branch 'immich-app:main' into feat/samsung-raw-and-fujifilm-raf

samip5 committed:
Remove un-used imports

samip5 committed:
The failing test.. is no longer failing.

samip5 committed:
Run prettier

samip5 committed:
Merge branch 'immich-app:main' into feat/samsung-raw-and-fujifilm-raf

samip5 committed:
Original implementation with just a catch block added.

alextran1502 commented on server/apps/immich/src/config/asset-upload.config.ts:
```diff
@@ -52,16 +50,20 @@ function fileFilter(req: Request, file: any, cb: any) {
-    logger.error(`Unsupported file type ${extname(file.originalname)} file MIME type ${file.mimetype}`);
-    cb(new BadRequestException(`Unsupported file type ${extname(file.originalname)}`), false);
-  }
+  // TODO: Create new API endpoint for mimetypes and use that here as browser's
```
Can we uncommented this?

samip5 replied:
Unfortunately with it uncommented, those files will fail to be uploaded due to the mimetype issue from browser.

samip5 replied:
Uncommented and added handling for the srw and raf formats even if browser reports them incorrectly.

alextran1502 commented on web/src/lib/utils/file-uploader.ts:
```diff
@@ -97,7 +93,12 @@ async function fileUploader(
 
 		// Get asset type
-		formData.append('assetType', assetType);
+		if (fileExtension.toLowerCase() == 'raf' || fileExtension.toLowerCase() == 'srw') {
```
Maybe a helper method here to check for raw file types?

samip5 replied:
Did that.

samip5 committed:
Merge branch 'immich-app:main' into feat/samsung-raw-and-fujifilm-raf

samip5 committed:
feat(server): Some tests and specific handling for the two raw formats

samip5 committed:
feat(web): Helper for raw image type.

samip5 committed:
Merge branch 'immich-app:main' into feat/samsung-raw-and-fujifilm-raf

samip5 committed:
Handling of mimetype on server

samip5 committed:
Handling of mimetypes on web with a map

samip5 committed:
Bring back the acceptedfile filter

samip5 committed:
Fix the asset-upload tests after changes

samip5 committed:
acceptedFile is not usable due to type being empty from browser.

samip5 committed:
Switch needs to use lowercase variants.

samip5 committed:
Address Discord comments

matthinc committed:
feat(mobile): Library page rework (album sorting, favorites) (#1501)

* Add album sorting

* Change AppBar to match photos page behaviour

* Add buttons

* First crude implementation of the favorites page

* Clean up

* Add favorite button

* i18n

* Add star indicator to thumbnail

* Add favorite logic to separate provider and fix favorite behavior in album

* Review feedback (Add isFavorite variable)

* dev: style buttons

* dev: styled drop down button

---------

Co-authored-by: Alex Tran <alex.tran1502@gmail.com>

martyfuhry committed:
feat(mobile): Tap to enter immersive mode on gallery viewer (#1546)

martyfuhry committed:
feat(mobile): Removed stay logged in checkbox and made it enabled by default (#1550)

* removed stay logged in checkbox and made it enabled by default

* adds padding to login button

* removed all isSaveLogin

* fix: logout would re-login with previous credential upon app restart

---------

Co-authored-by: Alex <alex.tran1502@gmail.com>

alextran1502 committed:
chore(server): remove token when logged out (#1560)

* chore(mobile): invoke logout() on mobile app

* feat: add mechanism to delete token from logging out endpoint

* fix: set state after login sequence success

* fix: not removing token when logging out from OAuth

* fix: prettier

* refactor: using accessTokenId to delete

* chore: pr comments

* fix: test

* fix: test threshold

EnochPrime committed:
feat(deployment): support docker secrets (#1254)

* Support secrets

* Rewrite to support sh

* Remove JWT_SECRET

martyfuhry committed:
fix(mobile): Added flutter native splash and splash screens (#1520)

* rebasing

* added launch background image to repository

---------

Co-authored-by: Marty Fuhry <marty@fuhry.farm>

zoodyy committed:
refactor(mobile): introduce Album & User classes (#1561)

replace usages of AlbumResponseDto with Album
replace usages of UserResponseDto with User

martyfuhry committed:
feat(mobile): Multiselect add to favorite from the timeline (#1558)

* multiselect add to favorites

alextran1502 committed:
feat(server): add updatedAt to Asset, Album and User (#1566)

* feat: add updatedAt info to DTO and generate api

* chore: remove unsued file

* chore: Add update statement to add/remove asset/user to album

* fix: test

zackpollard committed:
chore(server): update package-lock.json to match package.json (#1573)

alextran1502 committed:
chore(server) Add user FK to album entity (#1569)

dependabot[bot] committed:
chore(deps): bump docker/setup-buildx-action from 2.4.0 to 2.4.1 (#1575)

Bumps [docker/setup-buildx-action](https://github.com/docker/setup-buildx-action) from 2.4.0 to 2.4.1.
- [Release notes](https://github.com/docker/setup-buildx-action/releases)
- [Commits](https://github.com/docker/setup-buildx-action/compare/v2.4.0...v2.4.1)

---
updated-dependencies:
- dependency-name: docker/setup-buildx-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

alextran1502 committed:
chore(server): make owner as required response for AlbumResponseDto (#1579)

alextran1502 committed:
feat(GitHub): update bug and feature request template (#1584)

* dev: Reusing template from Home Assistant

* dev: add bug report template

* fix: template

* dev: change type

* dev:

* dev: add default labels

* dev: Add default title

* dev: add feature request template

* remove feature request from markdown

* dev: frontmatter

alextran1502 committed:
fix(GitHub): feature request template

alextran1502 committed:
fix(GitHub): feature request form has wrong type for textarea

martyfuhry committed:
feat(mobile): Responsive layout improvements with a navigation rail and album grid (#1583)

samip5 committed:
feat(proxy): Initial IPv6 support (#1577)

alextran1502 committed:
fix(server): Create album response doesn't have owner property as required (#1704)

michelheusschen committed:
feat(web): allow uploading more file types (#1570)

* feat(web): allow uploading more file types

* fix(web): make filename extension lowercase

alextran1502 commented:
@samip5 Can you rebase with main and keep the fallback implementation on generating thumbnails?

The frontend is handled by https://github.com/immich-app/immich/pull/1570. You can add the file type for RAF and SRW to the list of file name to infer its mimetype

zoodyy committed:
refactor(mobile): add Isar DB & Store class (#1574)

* refactor(mobile): add Isar DB & Store class

new Store: globally accessible key-value store like Hive (but based on Isar)

replace first few places of Hive usage with the new Store

* reduce max. DB size to prevent errors on older iOS devices

---------

Co-authored-by: Alex <alex.tran1502@gmail.com>

matthinc committed:
feat(mobile): Home screen customization options (#1563)

* Try staggered layout for home page

* Introduce setting for dynamic layout

* Fix some provider related bugs

* Make asset grouping configurable

* Add translation keys, refactor group title

* Rename enum values

* Fix enum names

* Reformat long if statement

* Fix timezone related bug

* Minor clean up

* Fix unit test

* Add second assets check back to home screen

alextran1502 committed:
[Localizely] Translations update (#1707)

alextran1502 committed:
fix(server): get shared link album info doesn't contain owner property (#1708)

Commit:
Version v1.46.0

samip5 committed:
Add raf and srw to the file names.

samip5 commented:
> Can you rebase with main and keep the fallback implementation on generating thumbnails?

Rebased.

samip5 committed:
Merge branch 'main' into feat/samsung-raw-and-fujifilm-raf

jrasm91 commented on server/apps/immich/src/config/asset-upload.config.ts:
```diff
@@ -48,13 +46,28 @@ export function customStorage(): StorageEngine {
 
+const logger = new Logger('AssetUploadConfig');
+
+function getMimeType(file: Express.Multer.File) {
+  const extension = file.originalname.split('.').pop() as string;
+  switch (extension.toLowerCase()) {
+    case 'raf':
+      return 'image/x-fuji-raf';
+    case 'srw':
+      return 'image/x-samsung-srw';
+    default:
+      return file.mimetype;
+  }
+}
+
```
I don't think we need this anymore since the web is sending the correct mimetype now, right?

Otherwise looks good.

samip5 replied:
Ah, you're right. Got rid of that. :)

samip5 replied:
Actually, it is still required.

samip5 committed:
Remember to add the extensions to fileSelector.

samip5 committed:
Removed the getMimeType function on server as shouldn't be needed anymore.

samip5 committed:
Revert "Removed the getMimeType function on server as shouldn't be needed anymore."

It is required still.

This reverts commit fc766dd0beeaa54eb27f73e4ec0bbe4913220fde.

samip5 committed:
Should use proper mimetypes.

samip5 commented:
@alextran1502 & @jrasm91 What's the status of this? What should I do to get it merged?

samip5 committed:
Merge branch 'immich-app:main' into feat/samsung-raw-and-fujifilm-raf

alextran1502 commented:
@samip5 Ah crap, I forgot about this one. Let me run some quick tests today and get it merged

samip5 commented:
> @samip5 Ah crap, I forgot about this one. Let me run some quick tests today and get it merged

It seems there's more merge conflicts again, gonna fix those and probably then it could be merged?

samip5 committed:
Merge branch 'main' into feat/samsung-raw-and-fujifilm-raf

alextran1502 commented:
@samip5  Can you help fix the linter issue so we can merge it in?

samip5 commented:
> @samip5 Can you help fix the linter issue so we can merge it in?

I had ran prettier, and it doesn't complain of any issues on my end.

alextran1502 commented:
@samip5 Here is the failed test case

![image](https://user-images.githubusercontent.com/27055614/218838261-ba4662ec-890d-47f9-a0bb-f741a51f4eda.png)

alextran1502 committed:
fix linter

alextran1502 commented:
@samip5 I helped you push the fix

samip5 commented:
Ah, my local prettier was not complaining at all despite that test.
